bokehjs/src/lib/core/has_props.ts

```diff
@@ -419,21 +421,23 @@ export abstract class HasProps extends Signalable() {
     }
   }
 
-  static _value_to_json(_key: string, value: unknown, _optional_parent_object: any): unknown {
+  static _value_to_json(value: unknown): unknown {
     if (value instanceof HasProps)
       return value.ref()
+    else if (is_NDArray(value))
```

Comment:
why mix snake and camel case? IMO `is_ndarray`, `encode_ndarray`, etc would be more consistent and preferable

Reply:
Because `NDArray` is supposed to be a class/type. Supposed, because it's not implemented yet due to technical limitations combined with backwards compatibility issues. Similarly, in `core/util/types` we have other type predicates like `isNumber` or `isArray`, which I intend to rename to `is_Number`, `is_Array`, etc., but the upper case syntax is to signify class/type as well.

Reply:
Absent any alternate meaning for `is_number`, `is_ndarray` that those might possibly be confused with, I think that convention is splitting hairs and unnecessary. I'd still prefer consistency but if you feel strongly I'll just leave it at -0